anise/src/astro/aberration.rs

```diff
@@ -0,0 +1,153 @@
+/*
+ * ANISE Toolkit
+ * Copyright (C) 2021-2023 Christopher Rabotin <christopher.rabotin@gmail.com> et al. (cf. AUTHORS.md)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Documentation: https://nyxspace.com/
+ */
+
+use core::f64::EPSILON;
+
+use crate::{
+    constants::SPEED_OF_LIGHT_KM_S,
+    errors::{AberrationSnafu, VelocitySnafu},
+    math::{rotate_vector, Vector3},
+};
+
+#[cfg(feature = "python")]
+use pyo3::prelude::*;
+use snafu::ensure;
+
+use super::PhysicsResult;
+
+/// Defines the available aberration corrections.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "python", pyclass)]
+#[cfg_attr(feature = "python", pyo3(module = "anise."))]
+pub enum Aberration {
```

Comment:
This is clunky and repetitive. Instead, I think this should be a simple struct with boolean fields `converged`, `light_time`, `transmission`. The combination of these three defines all nine combinations of the enum.

Reply:
Then there should be a `FromStr` to match the SPICE flags.